packages/core/src/validate/run.ts

```diff
@@ -241,13 +243,35 @@ function* getConstructorErrors(contractDef: ContractDefinition, decodeSrc: SrcDe
   }
 }
 
+function initCaches() {
+  const delegateCallCache: Cache = {
+    mainContractErrors: new Map<number, ValidationErrorOpcode[]>(),
+    inheritedContractErrors: new Map<number, ValidationErrorOpcode[]>(),
+    visitedNodeIds: new Map<number, Scope>(),
+  };
```

Comment:
We should avoid type annotations if they can be inferred. Here, because we're assigning to a variable that is explicitly annotatated as `Cache`, the type of the maps will be inferred from that.

If you change this function to `function initOpcodeCache(): OpcodeCache` then the return statement will also be able to infer the type automatically.

```suggestion
  const delegateCallCache: Cache = {
    mainContractErrors: new Map(),
    inheritedContractErrors: new Map(),
    visitedNodeIds: new Map(),
  };
```